aspnetcore/blazor/security/additional-scenarios.md

```diff
@@ -23,12 +23,231 @@ This article explains how to configure server-side Blazor for additional securit
+
+### Demonstration `Weather` component
+
+The following component is presented to demonstrate using an access token from the user's claims in C# code using an <xref:System.Net.Http.HttpRequestMessage>.
+
+> [!NOTE]
+> If the app uses [Microsoft identity platform](/entra/identity-platform/) with [Microsoft Identity Web packages/API](/entra/msal/dotnet/microsoft-identity-web/) for [Microsoft Entra ID](https://www.microsoft.com/security/business/microsoft-entra), we recommend using the API provided by Microsoft to call web APIs, which usually provides a better developer experience. The example in this section is only for demonstration purposes in an app that adopts Microsoft Identity Web for Entra.
+
+[CORS](xref:security/cors) configuration is required when the web API is hosted at a different origin than the calling app.
+
+With additional code, the weather forecast can be persisted when the component is prerendered. For more information, see <xref:blazor/components/prerender#persist-prerendered-state>.
+
+```razor
+@page "/weather"
+@using System.Net.Http.Headers
+@using Microsoft.AspNetCore.Authorization
+@using BlazorWebAppEntra.Client.Weather
+@attribute [Authorize]
+@inject IHttpClientFactory ClientFactory
+
+<PageTitle>Weather</PageTitle>
+
+<h1>Weather</h1>
+
+<p>This component demonstrates showing data.</p>
+
+@if (forecast == null)
+{
+    <p><em>Loading...</em></p>
+}
+else
+{
+    <table class="table">
+        <thead>
+            <tr>
+                <th>Date</th>
+                <th aria-label="Temperature in Celsius">Temp. (C)</th>
+                <th aria-label="Temperature in Fahrenheit">Temp. (F)</th>
+                <th>Summary</th>
+            </tr>
+        </thead>
+        <tbody>
+            @foreach (var forecast in forecast)
+            {
+                <tr>
+                    <td>@forecast.Date.ToShortDateString()</td>
+                    <td>@forecast.TemperatureC</td>
+                    <td>@forecast.TemperatureF</td>
+                    <td>@forecast.Summary</td>
+                </tr>
+            }
+        </tbody>
+    </table>
+}
+
+@code {
+    private IEnumerable<WeatherForecast>? forecast;
+
+    [CascadingParameter]
+    private Task<AuthenticationState>? AuthState { get; set; }
+
+    protected override async Task OnInitializedAsync()
+    {
+        if (AuthState is not null)
+        {
+            var authState = await AuthState;
+            var user = authState?.User;
+
+            if (user is not null)
+            {
+                var accessToken = 
+                    user.Claims.First(c => c.Type == "AccessToken")?.Value;
```

Comment:
I realize that since I'm going back on my suggestion to make an access token claim, we'll have to come up with another solution to get the access token during server interactive rendering. What about using IHttpContextAccessor in a CircuitHandler to store the access token in a scoped service? It seems like it'd be similar complexity of doing the same thing with PersistentComponentState but would be less likely to invite people with web assembly projects unwittingly exposing the wrong kind of access token to their public browser client.

Reply:
> What about using IHttpContextAccessor in a CircuitHandler to store the access token in a scoped service?

I'm not able to get the CircuitHandler approach to work. I think you'll have to give me the code to make it work, keeping in mind that I need it to work for Auto-rendering apps (CSR) ... but that's related to my next question ................

Could you explain this a bit more ...

> would be less likely to invite people with web assembly projects unwittingly exposing the wrong kind of access token to their public browser client.

Do you intend that devs should never pass an access token to an `Auto` component for use under CSR?

Reply:
> If the answer is 'yes' ... 😨 don't do that (unsafe middle-tier use) ... 

I think the answer is yes. We should never be passing an access token acquired by the server to the client for CSR. I think you're bullet points are fine. I can try to come up with a sample using a CircuitHandler, but I won't have time for that in the next couple weeks.